### PR TITLE
[Grid] made the grid adjustable

### LINKED ISSUE #76: Keep the fluid sperate from the pixel grid
I have seen your newest commit (4.0 pre 9) and i would like to point out that it is more convenient to keep the two grids seperate from each other. This way i can include or exclude (SASS) the different versions as needed. And it is clearer and more practical to adjust certain values if the two grids aren't merged together.

By the way i'm preparing some improvements for the grid and would like to contribute, are you ok with a pull request?

_Preview of my improvements:_ http://codepen.io/retober/pen/LNweww

### PR DESCRIPTION
**Pull request by retober**
I made the grid adjustable. Please have a look. Now it is possible to adjust the number of grid columns, the spacing between the columns and the initial grid width. You are also able to customize the breakpoints.

Closes #76

### CODE REVIEW

**Comment by catchamonkey** [comment]
Hey @retober thanks for the PR.
Can you sort out the indentation and formatting of your changes so it's easier to review please?

**Comment by retober** [comment]
I'm not used to this kind of formatting but i tried to fix the indentation.

File: scss/_grid.scss [hunk 1]
```diff
@@ -19,42 +40,20 @@ Screen 1200px+
...
-.grid--offset-3                                 { margin-left:25%; }
-.grid--offset-4                                 { margin-left:33.333%; }
-.grid--offset-5                                 { margin-left:41.667%; }
-.grid--offset-6                                 { margin-left:50%; }
-.grid--offset-7                                 { margin-left:58.333%; }
-.grid--offset-8                                 { margin-left:66.666%; }
-.grid--offset-9                                 { margin-left:75%; }
-.grid--offset-10                                { margin-left:83.333%; }
-.grid--offset-11                                { margin-left:91.667%; }
+@for $i from 1 through $grid-columns - 1 { .grid--offset-#{$i} { margin-left: 100% / $grid-columns *$i; } }
```
Comment (catchamonkey) [inline scss/_grid.scss]: do this one (and any others) as you did https://github.com/OwlyStuff/amazium/pull/80/files#diff-f6ca44772c5cdca97cf5be7357416fb8R47

That one is spot-on!

**Comment by catchamonkey** [comment]
Looks good so far @retober 

We will need to have an update to the docs too please.
Need to explain the process of building your own version if you want to override the grid quantity etc.
It might be best to break out the variables into a separate SCSS file, as that way to build your own you can just include your own setup var file, include the amazium files and generate your end result.
Do you see what I mean?

**Comment by retober** [comment]
I agree,  _variables.scss would be the best place to put the adjustable variables. What do you mean when you said that we need to update the docs. Which file do you mean ? Isn't amazium.co.uk the documentation ? I'm not a native speaker so i guess it is not advisable to let me write the documentation.

**Comment by catchamonkey** [comment]
@retober yes that's right, the amazium site covers all the usage and standard installation.
But the README in this repo also covers the basics, and think it would be best something relating to building your own column/gutter version in there.
Nothing wrong with your language at all, give it a go, we can always tweak it. But as this is your feature and approach it would be good to have you explain how you see it working.

Cheers 👍

**Comment by retober** [comment]
I'm writing a small documentation of my feature. There still seem to be some problems. And i have a question - do i have to add `.grid` now to every `.grid-X` or why does this class exist ?

**Comment by MikeBallan** [comment]
@retober You will only use .grid when you want all the columns in your row to be the same width. If you want a specified width you would then use .grid-6. Hope that makes sense.

**Comment by MikeBallan** [comment]
image

**Comment by MikeBallan** [comment]
Could do but grid is shorter to type :)

**Comment by retober** [comment]
I wrote a short documentation of the new feature and fixed some errors. I tested it and it works fine now but somebody else should have a look at it and test it thoroughly.

**Comment by p2or** [comment]
Thanks guys. Good job! What about calling it `.grid-x` instead of `.grid`?

**Comment by catchamonkey** [comment]
For me it's the default grid, without a size specific override (1-12), so just ```.grid``` makes sense.
Anything else would need to clarify it, which ```.grid-x``` doesn't do. Unless i've misunderstood what you meant with that one @p2or ?

**Comment by test2k** [comment]
Well done my friend! Rockin' in the Free World!

**Comment by catchamonkey** [comment]
Thanks for your work here @retober looking good!

I'll pull down and do some testing against various projects.
Docs look fine too so nice one there.

/cc @MikeBallan

**Comment by p2or** [comment]
@catchamonkey It's more about consistency. I agree that '.grid' is short and easy to write but I think having the same length of grid class names leads into a clean design. Also `.grid-x` or `grid-n` indicates that the class is supplied from the framework, it's more unique in a larger scope, but especially it implies that there is something automatic going on which is easier to understand (IMO).

**Comment by catchamonkey** [comment]
@p2or my issue with .```grid-x``` is that it doesn't add any clarity, as the **x** is as (possibly more) ambiguous as just ```.grid``` on it's own.

In terms of ensuring they are the same number of characters, that's kind of a side point, don't think that has ever been an issue in code clarity/cleanliness :)

**Comment by MikeBallan** [comment]
Had a chance to test this and looks great. Thanks for the PR!

**Pull request merged**